dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/DataRequestAlterationManager.kt

```diff
@@ -150,4 +156,33 @@ class DataRequestAlterationManager(
                 "reporting period ${metaData.reportingPeriod} and framework ${metaData.dataType.name}",
         )
     }
+
+    /**
+     * Method to patch data request corresponding to a dataset
+     * @param nonSourceableInfo the info on the non-sourceable dataset
```

Comment:
Because this function is only used in the processDataReportedNotSourceableMessage and it checks that isNonSouceable is true. I added an additional check in this function to check that isNonSourceable is indeed true.